src/main/webapp/js/utils/form/relation-field.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { getEntityArrayByName } from '../../selectors';
+import SelectField from './select-field';
+
+class RelationField extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: [{ label: null, value: null }],
+    };
+  }
+
+  componentDidMount() {
+    const { entity, fieldConfig } = this.props;
+    const { relatedField } = fieldConfig;
+    const options = _.map(entity, row => ({ label: row[relatedField], value: row.id }));
+    this.setState({ options });
+  }
+
+  render() {
+    const { key } = this.props;
+    let { fieldConfig } = this.props;
+    const { options } = this.state;
+    fieldConfig = {
+      ...fieldConfig,
```

Comment:
do this destructing directly when writing in SelectField prop, the you will not need to rewrite the fieldConfig variable